### PR TITLE
Pull images from Images pane

### LINKED ISSUE #1313: [Request] Pull image from images list
It would be great to be able to pull a newer version of an image from the list of images, either the latest tag if you select the image, or the specific tag if you select a specific tag in the image's list

### PR DESCRIPTION
**Pull request by bwateratmsft**
Allows pulling images from Images pane, without login. It was different enough from the existing pullImage from the Registries pane that I felt like it needed its own command.

Fixes #1313

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -352,20 +353,25 @@
                     "group": "images_2_general@1"
                 },
                 {
-                    "command": "vscode-docker.images.push",
```
Comment (bwateratmsft) [inline package.json:355]: The diff makes it hard to tell but this is actually inserting `pull` above `push` and re-indexing the others.

File: src/commands/registerCommands.ts [hunk 2]
```diff
@@ -138,7 +140,7 @@ export function registerCommands(): void {
     registerCommand('vscode-docker.context.use', useDockerContext);
     registerCommand('vscode-docker.context.inspect', inspectDockerContext);
     registerCommand('vscode-docker.context.remove', removeDockerContext);
-    registerCommand('vscode-docker.installDocker', installDocker);
+    registerWorkspaceCommand('vscode-docker.installDocker', installDocker);
```
Comment (bwateratmsft) [inline src/commands/registerCommands.ts:143]: Unrelated, I think this needs to be set up as a workspace command

Comment (ravipal) [reply src/commands/registerCommands.ts:143]: registerWorkspaceCommand ensured the command working on the local host?

Comment (bwateratmsft) [reply src/commands/registerCommands.ts:143]: If I understand it correctly, yes. I want to take some time to investigate and be sure.

File: src/commands/installDocker.ts [hunk 3]
```diff
@@ -5,10 +5,19 @@
 
 import * as os from 'os';
 import { IActionContext } from 'vscode-azureextensionui';
+import { localize } from '../localize';
+import { DockerExtensionKind, getVSCodeRemoteInfo } from '../utils/getVSCodeRemoteInfo';
 import { openExternal } from '../utils/openExternal';
 import { MacDockerInstaller, WindowsDockerInstaller } from './dockerInstaller';
 
 export async function installDocker(context: IActionContext): Promise<void> {
+    const remoteInfo = getVSCodeRemoteInfo(context);
```
Comment (bwateratmsft) [inline src/commands/installDocker.ts:14]: @ravipal I looked into `registerWorkspaceCommand`, it's essentially made to warn you if you are in UI mode but try to do something like `docker build`, which must always run in the workspace. So, it doesn't do what we need.

Instead I put this block in; if remoting is in use then we definitely don't want to get into the business of trying to install Docker (plus the target of remoting would almost always be Linux where we don't have support anyway).

Can you take a look at this code and see if it looks OK?

Comment (ravipal) [reply src/commands/installDocker.ts:14]: Can this check be added during register command? something like `registerLocalCommand` so in future other such commands can reuse the code.

Comment (bwateratmsft) [reply src/commands/installDocker.ts:14]: Good idea! I'll do that.

**Pull request merged**